addons/sale/views/product_attribute_views.xml

```diff
@@ -32,11 +32,12 @@
         <field name="priority" eval="8"/>
         <field name="arch" type="xml">
             <xpath expr="//field[@name='create_variant']" position="before">
-                <field name="type" widget="radio"/>
+                <field name="display_type" widget="radio"/>
             </xpath>
             <xpath expr="//field[@name='value_ids']/tree/field[@name='name']" position="after">
-                <field name="is_custom" groups="product.group_product_variant" />
-                <field name="html_color"/>
+                <field name="display_type" invisible="1"/>
+                <field name="is_custom" groups="product.group_product_variant"/>
+                <field name="html_color" attrs="{'column_invisible': [('parent.display_type', '!=', 'color')]}" widget="color"/>
```

Comment:
@Arcasias This widget opens the color picker in readonly view and conflict with the form view. The behavior is ok in edit view.

Reply:
@arj-odoo PR for the fix: https://github.com/odoo/odoo/pull/36949
It is currently being reviewed.